### PR TITLE
Add new FHIR library with R4 support

### PR DESCRIPTION
**Pull request by wmontwe**
## Description

Bumped:
* Android Studio 4.0.0 -> 4.1.1 and added necessary config changes
* Gradle 6.5 -> 6.7.1 for performance reasons
* FHIR SDK 0.7.0 -> 1.0.0 -> [BREAKING change](https://github.com/d4l-data4life/hc-fhir-sdk-java/releases/tag/v1.0.0)
* FHIR Helper SDK 1.3.1 -> 1.4.0 -> [BREAKING change](https://github.com/d4l-data4life/hc-fhir-helper-sdk-kmp/releases/tag/v1.4.0)

Removed codeowners file as it's breaking the automatic reviewers assignment.

## Motivation and Context
Updated FHIR libraries and project to prepare extension of the SDKs API to support FHIR R4, which will be addressed in a followup pull-request.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sample-android/build.gradle.kts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 plugins {
     id("com.android.application")
     id("kotlin-android")
-    id("com.getkeepsafe.dexcount")
+//    id("com.getkeepsafe.dexcount")
```
Comment (bitPogo) [inline sample-android/build.gradle.kts:20]: Nitpick: Is that any longer needed?

**Review by bitPogo — COMMENTED** [review]
One tiny nitpick...besides that looks good to me

File: sample-android/build.gradle.kts (hunk 1 above)
Comment (wmontwe) [reply sample-android/build.gradle.kts:20]: it is, but need to wait for an update of the library

Comment (bitPogo) [reply sample-android/build.gradle.kts:20]: Ok

**Review by sebastianwoinar — COMMENTED** [review]
Looks good!

Do you think that additional examples for R4 would bring any value?

**Comment by wmontwe** [comment]
> Looks good!
> 
> Do you think that additional examples for R4 would bring any value?

So far this is only adding the new FHIR library and do the necessary migrations for existing FHIR 3 usage. FHIR 4 support is added in another pull-request

**Review by bitPogo — APPROVED** [review]
Looks good to me

**Pull request merged**